### PR TITLE
JBPM-5660: Cloning repositories is allowed even if create repositories permission is disabled

### PR DESCRIPTION
**Pull request by dgutierr**
Fix + test.

@manstis Can you review it please?

### CODE REVIEW

File: kie-drools-wb-parent/kie-drools-wb-webapp/src/main/java/org/kie/workbench/drools/client/perspectives/AdministrationPerspective.java [hunk 1]
```diff
@@ -122,29 +122,24 @@ public Menus getMenus() {
...
-        this.cloneRepoCommand = new Command() {
+    public Command getNewRepoCommand() {
+        return newRepoCommand;
+    }
 
-            @Override
-            public void execute() {
-                cloneRepositoryPresenter.showForm();
-            }
+    public Command getCloneRepoCommand() {
```
Comment (manstis) [inline kie-drools-wb-parent/kie-drools-wb-webapp/src/main/java/org/kie/workbench/drools/client/perspectives/AdministrationPerspective.java]: Does this have to be ```public```? (package-protected would be fine.. but just a detail).

File: kie-drools-wb-parent/kie-drools-wb-webapp/src/main/java/org/kie/workbench/drools/client/perspectives/AdministrationPerspective.java [hunk 2]
```diff
@@ -122,29 +122,24 @@ public Menus getMenus() {
                 .endMenu().build();
     }
 
-    private void buildCommands() {
-        this.cloneRepoCommand = new Command() {
+    public Command getNewRepoCommand() {
```
Comment (manstis) [inline kie-drools-wb-parent/kie-drools-wb-webapp/src/main/java/org/kie/workbench/drools/client/perspectives/AdministrationPerspective.java]: Does this have to be ```public```? (package-protected would be fine.. but just a detail).

File: kie-drools-wb-parent/kie-drools-wb-webapp/src/test/java/org/kie/workbench/drools/client/AdministrationPerspectiveTest.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
```
Comment (manstis) [inline kie-drools-wb-parent/kie-drools-wb-webapp/src/test/java/org/kie/workbench/drools/client/AdministrationPerspectiveTest.java]: 2017

File: kie-wb-parent/kie-wb-webapp/src/main/java/org/kie/workbench/client/perspectives/AdministrationPerspective.java [hunk 4]
```diff
@@ -122,29 +122,23 @@ public Menus getMenus() {
                 .endMenu().build();
     }
 
-    private void buildCommands() {
-        this.cloneRepoCommand = new Command() {
+    public Command getNewRepoCommand() {
```
Comment (manstis) [inline kie-wb-parent/kie-wb-webapp/src/main/java/org/kie/workbench/client/perspectives/AdministrationPerspective.java]: Does this have to be ```public```? (package-protected would be fine.. but just a detail).

File: kie-wb-parent/kie-wb-webapp/src/main/java/org/kie/workbench/client/perspectives/AdministrationPerspective.java [hunk 5]
```diff
@@ -122,29 +122,23 @@ public Menus getMenus() {
...
+        return newRepoCommand;
+    }
 
-            @Override
-            public void execute() {
-                cloneRepositoryPresenter.showForm();
-            }
+    public Command getCloneRepoCommand() {
```
Comment (manstis) [inline kie-wb-parent/kie-wb-webapp/src/main/java/org/kie/workbench/client/perspectives/AdministrationPerspective.java]: Does this have to be ```public```? (package-protected would be fine.. but just a detail).

File: kie-wb-parent/kie-wb-webapp/src/test/java/org/kie/workbench/client/AdministrationPerspectiveTest.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
```
Comment (manstis) [inline kie-wb-parent/kie-wb-webapp/src/test/java/org/kie/workbench/client/AdministrationPerspectiveTest.java]: 2017

**Comment by dgutierr** [comment]
@manstis Changes done. Thanks for the feedback!

**Comment by manstis** [comment]
Thanks for the changes @dgutierr

**Comment by tomasdavidorg** [comment]
@dgutierr, thanks for the fix.

**Pull request merged**